views/xTree/XTreeView.js

```diff
@@ -547,7 +549,20 @@ window.XTreeAncestorList = class XTreeAncestorList {
         }
 
         for (const index in ancestorData) {
-            thePeopleList.add(ancestorData[index]);
+            let thePerson = ancestorData[index];
+            if (thePerson.Id < 0) {
+                thePerson.Id = 100 - thePerson.Id;
```

Comment:
I have not looked in detail how/when this code is called, but if `thePeopleList` is global and you may be adding to it as result of more than one `getPeople` call, then you will be overwriting people here (i.e. regard them as the same person) even though they are not. Or at least, you will be doing that until the getPeople API's handling of private profiles changes. (Or has it already been changed?). Currently (or up to very recently) private profiles are numbered from -1 in every result. i.e. if you make call 1, and there are private profiles, they will be numbered -1, -2, etc. When you make call 2 (even if it is just a paging call to get the next set of 1000 profiles), any private profiles will be numbered starting from -1 again.